Assets/BossRoom/Scripts/Client/Game/Action/MeleeActionFX.cs

```diff
@@ -44,7 +45,17 @@ public override bool Start()
             {
                 float padRange = Description.Range + k_RangePadding;
 
-                if ((m_Parent.transform.position - targetNetworkObj.transform.position).sqrMagnitude < (padRange * padRange))
+                Vector3 targetPosition;
+                if (PhysicsWrapper.TryGetPhysicsWrapper(Data.TargetIds[0], out var physicsWrapper))
+                {
+                    targetPosition = physicsWrapper.Transform.position;
+                }
+                else
+                {
+                    targetPosition = targetNetworkObj.transform.position;
+                }
+
+                if ((m_Parent.transform.position - targetPosition).sqrMagnitude < (padRange * padRange))
```

Comment:
On a stylistic note - I would much rather prefer this boolean value to be captured into a variable with an explicit name VS having a comment. Has to do with comments being fragile code smells almost in all cases.

Reply:
It's not a request to change anything - more so a general observation about our codebase. We have these comments that describe what's happening in the logic, where a properly placed and named variable would do a better job in the readability and maintainability department.

Reply:
Agree on this. Best to just be explicit.